serve.go

```diff
@@ -80,54 +88,44 @@ func (h healthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	log.Printf("%v: requesting health probe", r.RemoteAddr)
 	resp := h.processAndEnsureCached(negotiatedContentType, outputer)
 	w.Header().Set(http.CanonicalHeaderKey("Content-Type"), negotiatedContentType)
-	w.WriteHeader(resp.statusCode)
+	statusCode := resp.toHTTPStatus()
+	w.WriteHeader(statusCode)
 	logBody := ""
-	if resp.statusCode != http.StatusOK {
+	if statusCode != http.StatusOK {
+		// if there are any test failures, log all the details since machine state is volatile
```

Comment:
This comment was a driveby addition, since we talked about output verbosity within the PR.